ui/src/org/pentaho/di/ui/core/database/wizard/CreateDatabaseWizardPageJDBC.java

```diff
@@ -163,39 +168,58 @@ public void modifyText( ModifyEvent arg0 ) {
 
   public void setData() {
     wHostname.setText( Const.NVL( databaseMeta.getHostname(), "" ) );
-
     wPort.setText( databaseMeta.getDatabasePortNumberString() );
 
-    wDBName.setText( Const.NVL( databaseMeta.getDatabaseName(), "" ) );
+    if ( !defaultWebAppNameSet && DATA_SERVICES_PLUGIN_ID.equals( databaseMeta.getPluginId() ) ) {
+      wDBName.setText( DEFAULT_WEB_APPLICATION_NAME );
+      defaultWebAppNameSet = true;
+    } else {
+      wDBName.setText( Const.NVL( databaseMeta.getDatabaseName(), "" ) );
+    }
+
+    if ( DATA_SERVICES_PLUGIN_ID.equals( databaseMeta.getPluginId() ) ) {
+      wlDBName.setText( BaseMessages.getString( PKG, "CreateDatabaseWizardPageJDBC.WebAppName.Label" ) );
+    } else {
+      wlDBName.setText( BaseMessages.getString( PKG, "CreateDatabaseWizardPageJDBC.DBName.Label" ) );
+    }
   }
 
   public boolean canFlipToNextPage() {
     String server =
-      wHostname.getText() != null ? wHostname.getText().length() > 0 ? wHostname.getText() : null : null;
+        wHostname.getText() != null ? wHostname.getText().length() > 0 ? wHostname.getText() : null : null;
     String port = wPort.getText() != null ? wPort.getText().length() > 0 ? wPort.getText() : null : null;
     String dbname = wDBName.getText() != null ? wDBName.getText().length() > 0 ? wDBName.getText() : null : null;
 
-    if ( server == null || port == null || dbname == null ) {
+    if ( ( server == null || port == null || dbname == null ) && !DATA_SERVICES_PLUGIN_ID
+        .equals( databaseMeta.getPluginId() ) ) {
       setErrorMessage( BaseMessages.getString( PKG, "CreateDatabaseWizardPageJDBC.ErrorMessage.InvalidInput" ) );
+
+      return false;
+    }
+
+    if ( ( server == null || port == null ) && DATA_SERVICES_PLUGIN_ID.equals( databaseMeta.getPluginId() ) ) {
```

Comment:
Small point (ignore it if you want)-- you've repeated "DATA_SERVICES_PLUGIN_ID.equals(databaseMeta.getPluginId() )" 4 or 5 times.  Might be more readable as a ".isDataServices( databaseMeta )" method.